## This a illustration of a Kenya imports on wine for the 2024

# Assignment 2
### Find the Rolling Mean for the temperatures over the  course of the Months from the weather data

In [26]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [51]:
import sys

print(sys.executable)
print(sys.version)

c:\Users\PatrickNjuguna\Desktop\Wine_importsdata\.venv-1\Scripts\python.exe
3.13.14 (tags/v3.13.14:fd17997, Jun 10 2026, 13:03:48) [MSC v.1944 64 bit (AMD64)]


In [28]:
import pandas as pd

df = pd.read_csv(
    r"C:\Users\PatrickNjuguna\Desktop\WITS-By-HS6Product.csv",
    delimiter=",",
    encoding="utf-8"
)

In [29]:
df.shape

(46, 9)

In [30]:
df.head()
df.shape
df.columns
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 46 entries, 0 to 45
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Reporter             46 non-null     str    
 1   TradeFlow            46 non-null     str    
 2   ProductCode          46 non-null     int64  
 3   Product Description  46 non-null     str    
 4   Year                 46 non-null     int64  
 5   Partner              46 non-null     str    
 6   Trade Value 1000USD  46 non-null     float64
 7   Quantity             46 non-null     float64
 8   Quantity Unit        46 non-null     str    
dtypes: float64(2), int64(2), str(5)
memory usage: 3.4 KB


In [31]:
df.describe# statistical numerics of columns and rows


<bound method NDFrame.describe of    Reporter TradeFlow  ProductCode  \
0     Kenya    Import         2204   
1     Kenya    Import         2204   
2     Kenya    Import         2204   
3     Kenya    Import         2204   
4     Kenya    Import         2204   
5     Kenya    Import         2204   
6     Kenya    Import         2204   
7     Kenya    Import         2204   
8     Kenya    Import         2204   
9     Kenya    Import         2204   
10    Kenya    Import         2204   
11    Kenya    Import         2204   
12    Kenya    Import         2204   
13    Kenya    Import         2204   
14    Kenya    Import         2204   
15    Kenya    Import         2204   
16    Kenya    Import         2204   
17    Kenya    Import         2204   
18    Kenya    Import         2204   
19    Kenya    Import         2204   
20    Kenya    Import         2204   
21    Kenya    Import         2204   
22    Kenya    Import         2204   
23    Kenya    Import         2204   
24    Kenya    I

In [32]:
df[["Year", "Partner", "Trade Value 1000USD", "Quantity"]].describe

<bound method NDFrame.describe of     Year                        Partner  Trade Value 1000USD    Quantity
0   2024                          World             22686.92  8055030.00
1   2024                   South Africa             10196.96  5017750.00
2   2024                         France              4326.46   574321.00
3   2024                          Italy              2645.45   643282.00
4   2024                          Spain              1310.18   516496.00
5   2024                          Chile              1182.54   392658.00
6   2024                      Australia               735.26   185602.00
7   2024                      Argentina               646.17   124823.00
8   2024                         Zambia               371.96   170122.00
9   2024                    New Zealand               284.02    43980.40
10  2024                        Moldova               280.92   145094.00
11  2024                  United States               176.82    32305.80
12  2024         

In [33]:
df[["Year","Partner","Trade Value 1000USD", "Quantity"]]

,Year,Partner,Trade Value 1000USD,Quantity
0,2024,World,22686.92,8055030.00
1,2024,South Africa,10196.96,5017750.00
2,2024,France,4326.46,574321.00
3,2024,Italy,2645.45,643282.00
4,2024,Spain,1310.18,516496.00
5,2024,Chile,1182.54,392658.00
6,2024,Australia,735.26,185602.00
7,2024,Argentina,646.17,124823.00
8,2024,Zambia,371.96,170122.00
9,2024,New Zealand,284.02,43980.40


In [34]:
for column in df.columns:
    print(column)

Reporter
TradeFlow
ProductCode
Product Description
Year
Partner
Trade Value 1000USD
Quantity
Quantity Unit


In [35]:
df.columns = [
    "Reporter",
    "Trade_flow",
    "Product_code",
    "Product_description",
    "Year",
    "Partner",
    "Trade value",
    "Quantity",
    "Quantity_unit"
]

In [36]:
df.columns

Index(['Reporter', 'Trade_flow', 'Product_code', 'Product_description', 'Year',
       'Partner', 'Trade value', 'Quantity', 'Quantity_unit'],
      dtype='str')

In [37]:
df["Year"].unique()

array([2024])

In [38]:
df['Year'].value_counts().sort_index()

Year
2024    46
Name: count, dtype: int64

In [39]:

df["Partner"].value_counts().sort_index()

Partner
 World                           1
Afghanistan                      1
Argentina                        1
Armenia                          1
Australia                        1
Austria                          1
Bosnia and Herzegovina           1
Chile                            1
China                            1
Denmark                          1
France                           1
Georgia                          1
Germany                          1
Hungary                          1
India                            1
Israel                           1
Italy                            1
Japan                            1
Jordan                           1
Lebanon                          1
Mexico                           1
Moldova                          1
Namibia                          1
Nepal                            1
Netherlands                      1
New Zealand                      1
North Macedonia                  1
Philippines                      1
Portugal    

In [40]:
for col in df.columns:
    print(repr(col))

'Reporter'
'Trade_flow'
'Product_code'
'Product_description'
'Year'
'Partner'
'Trade value'
'Quantity'
'Quantity_unit'


In [41]:
df.groupby("Partner")["Trade value"].sum().sort_values(ascending=False)

Partner
 World                           22686.92
South Africa                     10196.96
France                            4326.46
Italy                             2645.45
Spain                             1310.18
Chile                             1182.54
Australia                          735.26
Argentina                          646.17
Zambia                             371.96
New Zealand                        284.02
Moldova                            280.92
United States                      176.82
Portugal                           113.56
United Kingdom                     102.48
Germany                             73.47
Tanzania                            48.43
Ukraine                             38.17
Namibia                             26.36
Netherlands                         23.19
North Macedonia                     22.20
Israel                              15.82
Singapore                           11.77
Mexico                              10.71
United Arab Emirates      

In [42]:
df_countries = df["Partner"].str.strip()#removing the world from main dta frame and moving it to another

In [43]:
df["Partner"] = df["Partner"].str.strip()#removing extrace spaces

In [44]:
(df["Partner"].iloc[0])

'World'

In [45]:
df_countries = df[df["Partner"] != "World"]

In [46]:
df_countries["Partner"].head()

1    South Africa
2          France
3           Italy
4           Spain
5           Chile
Name: Partner, dtype: str

In [47]:
df_countries["Partner"].value_counts()

Partner
South Africa                     1
France                           1
Italy                            1
Spain                            1
Chile                            1
Australia                        1
Argentina                        1
Zambia                           1
New Zealand                      1
Moldova                          1
United States                    1
Portugal                         1
United Kingdom                   1
Germany                          1
Tanzania                         1
Ukraine                          1
Namibia                          1
Netherlands                      1
North Macedonia                  1
Israel                           1
Singapore                        1
Mexico                           1
United Arab Emirates             1
Hungary                          1
Austria                          1
India                            1
Serbia, FR(Serbia/Montenegro)    1
Japan                            1
Romania     

## Each countries % for imports to kenya

In [48]:
df_countries = df[df["Partner"] != "World"]

total_import_value = df_countries["Trade value"].sum()

country_share = (
    df_countries.groupby("Partner")["Trade value"]
    .sum()
    / total_import_value
    * 100
).sort_values(ascending=False)

country_share.head(15)


Partner
South Africa      44.946348
France            19.070250
Italy             11.660663
Spain              5.775036
Chile              5.212422
Australia          3.240893
Argentina          2.848200
Zambia             1.639532
New Zealand        1.251909
Moldova            1.238244
United States      0.779390
Portugal           0.500552
United Kingdom     0.451713
Germany            0.323842
Tanzania           0.213471
Name: Trade value, dtype: float64

In [49]:
country_analysis = (
    df_countries
    .groupby("Partner")
    .agg(
        trade_value=("Trade value", "sum"),
        quantity=("Quantity", "sum")
    )
    .sort_values("trade_value", ascending=False)
)

## 100 is the exchange rate

In [50]:
total_import_value = country_analysis["trade_value"].sum()
country_analysis["share_%"] =(
    country_analysis["trade_value"]/ total_value * 100
)
total_import_value


NameError: name 'total_value' is not defined

### South Africa is the largest partner by import value, contributing about 44.9% of the country-level import value in this 2024 dataset.
France is second at about 19.1%.
Italy is third at about 11.7%.
The top five — South Africa, France, Italy, Spain and Chile — account for roughly 86.7% of the country-level import value.
Quantity and value tell different stories. For example, South Africa supplies much more volume, while France has a much higher value relative to its litres.
This suggests that wine from different countries has substantially different reported import values per litre.
The dataset is for 2024 only, so we can describe the 2024 import pattern but cannot yet say whether imports are increasing or decreasing over time.


In [ ]:
country_analysis.head()

### This help determines value per litre imported from each country
| Measure             | What it tells us                                        |
| ------------------- | ------------------------------------------------------- |
| **Trade value**     | Which countries account for the most import expenditure |
| **Quantity**        | Which countries supply the most wine                    |
| **Value per litre** | The reported import value associated with each litre    |


In [ ]:
country_analysis["value_per_litre"] =(
    country_analysis["trade_value"]*1000 /country_analysis["quantity"]
)
country_analysis.sort_values( "value_per_litre",ascending = False)
country_analysis.head(20)

### Culculate overall average value
total_value_per_litre = (
    country_analysis["trade_value"].sum() * 1000
    / country_analysis["quantity"].sum()
)

total_value_per_litre

In [ ]:
total_value_per_litre =(
    country_analysis["trade_value"].sum() *1000 
    / country_analysis["quantity"].sum()
)
total_value_per_litre

### average value per litre

In [ ]:
country_analysis["vs_average'"] =( country_analysis["value_per_litre"] - total_value_per_litre)
country_analysis

create the average ["vs_average"]

In [ ]:
country_analysis = country_analysis.rename(
    columns={"vs_average'": "vs_average"}
)

In [ ]:
country_analysis["vs_average"] = (
    country_analysis["value_per_litre"]
    - total_value_per_litre
)

In [ ]:
country_analysis = country_analysis.loc[
    :, ~country_analysis.columns.duplicated()
]

In [ ]:
country_analysis.columns.tolist()

In [ ]:
country_analysis[ ["value_per_litre", "vs_average"]
].sort_values(
    "value_per_litre",
    ascending=False
)
country_analysis.head(10)

### How dependent is Kenya's wine import market on a small number of supplying countries?

In [ ]:
top_5_share= country_analysis.head(5)["share_%"].sum()


In [ ]:
top_10_share = country_analysis.head(10)["share_%"].sum()

In [ ]:
total_quantity = country_analysis["quantity"].sum()

country_analysis["_share_%"]= ( country_analysis["quantity"]/ total_quantity*100)

In [ ]:
total_quantity = country_analysis["quantity"].sum()

country_analysis["quantity_share_%"] = (
    country_analysis["quantity"] / total_quantity * 100
)

## Investigating the share of what has been imported in yr2024 by country

In [ ]:
country_analysis[
    ["quantity", "quantity_share_%"]].sort_values(    "quantity",ascending=False).head(10)

In [ ]:
import matplotlib.pyplot as plt

top_10 = (
    country_analysis
    .sort_values("quantity", ascending=False)
    .head(10)
)

plt.figure(figsize=(8, 8))

plt.pie(
    top_10["quantity"],
    labels=top_10.index,
    autopct="%1.1f%%",
    startangle=90
)
plt.xticks(rotation=45, fontsize=7)
plt.title("Top 10 Wine Import Suppliers to Kenya by Quantity - 2024")
# Key
plt.bar([], [], color="blue", label="High: > 2,000 L")
plt.bar([], [], color="green", label="Medium: 300,000–1,000,000 L")
plt.bar([], [], color="orange", label="Low: < 300,000 L")

plt.legend(title="Quantity category")
plt.legend(
    title="Quantity category",
    fontsize=7,
    title_fontsize=8
)

plt.show()

In [ ]:
import matplotlib.pyplot as plt

top_10 = (
    country_analysis
    .sort_values("quantity", ascending=False)
    .head(10)
)

plt.figure(figsize=(15, 8))

plt.bar(
       top_10.index,
    top_10["quantity"])

plt.title("Top 10 Wine Import Suppliers to Kenya by Quantity - 2024")

plt.show()


This answers: Among countries supplying a meaningful volume, whose imported wine has the highest reported value per litre?

In [ ]:
country_analysis[
    ["quantity", "value_per_litre"]
].sort_values(
    "value_per_litre",
    ascending=False
).head(10)## countries like Uganda and Netherlands has the highes value per liter but has very litre volume

In [ ]:
country_analysis[
    country_analysis["quantity"] >= 10000
][
    ["quantity", "value_per_litre"]
].sort_values(
    "value_per_litre",
    ascending=False
).head(10)

In [ ]:
top_10_share= (
    country_analysis
    .sort_values("trade_value", ascending=False)
    .head(10)
)

plt.figure(figsize=(8, 7))

plt.pie(
    top_10["trade_value"],
    labels=top_10.index,
    autopct="%1.1f%%",
    startangle=90
)

plt.title("Top 10 Wine Imports Countries by Value - Kenya, 2024")

plt.show()